### DISCUSSION TITLE
How to use different exogenous features in column ensembles?

### DISCUSSION THREAD

**Discussion by fkiraly** [alan-turing-institute/sktime#3731] [Q&A]
From slack:

How do I specify which exogenous features to use for each column in the column ensembleforecaster?
I am forecasting demand for 150 products, for multiple countries, on multiple time scales (day/week) and multiple levels (warehouse/customer).

**Comment by fkiraly** [comment]
You can do this by using pipelines as forecasters on the columns.

You need to use a `ForecastingPipeline` with a `ColumnSelect` transformer, this selects columns on the exogeneous data.

For example, let's say we have variables `"foo"` and `"bar"` on `X`, and "a" and "b" on "y", and two forecasters `f1`, `f2`, and want to use `"foo"` in `X` in `f1`, and both variables `"foo"`, `"bar"` in `X` in `f2`, you would do as follows:

```python
from sktime.forecasting.compose import ColumnEnsembleForecaster
from sktime.transformations.series.subset import ColumnSelect

forecasters = [
    ("name1", ColumnSelect("foo") ** f1, "a"),
    ("name2", f2, "b"),
]
col_ens = ColumnEnsembleForecaster(forecasters)
```

(the `**` is shorthand for creating a `ForecastingPipeline`)

**Comment by RikStarmans** [comment]
Thanks @Franz Király, I can confirm it works. I convert my column encoding to integer as sktime does not seem to support multi-level columns.
```python
from sktime.forecasting.statsforecast import StatsForecastAutoARIMA

forecasters = []
for product in products:
    for warehouse in warehouses:
        forecasters.append((f"{warehouse}_{product}",
                            ColumnSelect(feature_data.columns.get_loc(warehouse)) **
                            StatsForecastAutoARIMA(),
                            (label_data.columns
                                       .get_loc((product, warehouse)))))
pipe = make_pipeline(ColumnEnsembleForecaster(forecasters))
pipe.fit(X=feature_data, y=label_data)
```
